features/all-positions/AllPositions.tsx

```diff
@@ -68,49 +66,53 @@ const AllPositions = () => {
         </Typography>
       </Box>
       <Box py={4}>
-        <TableContainer component={Paper}>
-          <Table>
-            <TableHead>
-              <TableRow>
-                <TableCell>Active Sponsor</TableCell>
-                <TableCell align="right">Locked Collateral</TableCell>
-                <TableCell align="right">Tokens Outstanding</TableCell>
-                <TableCell align="right">
-                  Collateral Ratio (using price:{" "}
-                  {latestPrice ? utils.formatEther(latestPrice) : "N/A"})
-                </TableCell>
-              </TableRow>
-            </TableHead>
-            <TableBody>
-              {activeEmpSponsors &&
-                Object.keys(activeEmpSponsors).map((sponsor: string) => (
-                  <TableRow key={sponsor}>
-                    <TableCell component="th" scope="row">
-                      {sponsor}
-                    </TableCell>
-                    <TableCell align="right">
-                      {prettyBalance(
-                        activeEmpSponsors[sponsor].lockedCollateral
-                      )}
-                    </TableCell>
-                    <TableCell align="right">
-                      {prettyBalance(
-                        activeEmpSponsors[sponsor].tokensOutstanding
-                      )}
-                    </TableCell>
-                    <TableCell align="right">
-                      {prettyBalance(
-                        getCollateralRatio(
-                          activeEmpSponsors[sponsor].lockedCollateral,
-                          activeEmpSponsors[sponsor].tokensOutstanding
-                        )
-                      )}
-                    </TableCell>
-                  </TableRow>
-                ))}
-            </TableBody>
-          </Table>
-        </TableContainer>
+        {activeSponsors && (
+          <TableContainer component={Paper}>
+            <Table>
+              <TableHead>
+                <TableRow>
+                  <TableCell>Active Sponsor</TableCell>
+                  <TableCell align="right">Locked Collateral</TableCell>
+                  <TableCell align="right">Tokens Outstanding</TableCell>
+                  <TableCell align="right">
+                    Collateral Ratio (using price:{" "}
+                    {latestPrice ? latestPrice : "N/A"})
+                  </TableCell>
+                </TableRow>
+              </TableHead>
+              <TableBody>
+                {Object.keys(activeSponsors).map((sponsor: string) => {
+                  const activeSponsor = activeSponsors[sponsor];
+                  return (
+                    activeSponsor &&
+                    activeSponsor.collateral &&
+                    activeSponsor.tokensOutstanding && (
```

Comment:
```suggestion
                    activeSponsor?.collateral &&
                    activeSponsor?.tokensOutstanding && (
```

I think this is actually sufficient to pass typecheck.